### ISSUE TITLE
FunctionExpression navigation

### ISSUE THREAD

**Issue by reda-alaoui** [dsherret/ts-simple-ast#120]
Similarly to `FunctionDeclaration`, in a `FunctionExpression`, I would like to retrieve the:
- name
- body
- parameters
- return type

Example:
```javascript
const helloWorld = function(){
  // Do something
}
```

I found a workaround using untyped children but it is not optimal.

Could it be implemented?

**Comment by dsherret** [comment]
Yes, I could implement some useful helper functions on the initializer. `.getName()` would probably only work for a named function expression though:

```typescript
const helloWorld = function someName() {
  // Do something
};
```

If someone wanted the name in the original example they would have to get it from the variable declaration.

**Comment by dsherret** [comment]
@reda-alaoui This is available in 0.88.0. I didn't test it much though because I'm assuming the mixins that it uses will work with it. Let me know if you run into any issues.

By the way, in the scenario you gave, you can get the `FunctionExpression` by doing the following:

```typescript
// untested code, but it should show the main idea
import Ast, {FunctionExpression, TypeGuards} from "ts-simple-ast";

// ...

const helloWorldConst = sourceFile.getVariableDeclarationOrThrow("helloWorld");
helloWorldConst.getName(); // helloWord
const initializer = helloWorldConst.getInitializerOrThrow();
const functionExpr = initializer as FunctionExpression;
functionExpr.getParameters(); // []
functionExpr.getReturnType(); // void

// or for an initializer, you can check if it's a function expression using the type guard:
if (TypeGuards.isFunctionExpression(initializer)) {
    initializer.getParameters(); // []
    // etc...
}
```
